cnd/migrations/2020-04-09-030124_create-message-tables/up.sql

```diff
@@ -0,0 +1,110 @@
+    hash_function            NOT NULL,
+    bitcoin_refund_identity  NOT NULL,
+    ethereum_redeem_identity NOT NULL,
+    bitcoin_expiry           NOT NULL,
+    ethereum_expiry          NOT NULL,
+    secret_hash              NOT NULL
+);
+
+CREATE TABLE rfc003_ethereum_bitcoin_ether_bitcoin_request_messages
+(
+    id INTEGER               NOT NULL PRIMARY KEY,
+    swap_id UNIQUE           NOT NULL,
+    bitcoin_network          NOT NULL,
+    ethereum_chain_id        NOT NULL,
+    bitcoin_amount           NOT NULL,
+    ether_amount             NOT NULL,
+    hash_function            NOT NULL,
+    bitcoin_redeem_identity  NOT NULL,
+    ethereum_refund_identity NOT NULL,
+    bitcoin_expiry           NOT NULL,
+    ethereum_expiry          NOT NULL,
+    secret_hash              NOT NULL
+);
+
+CREATE TABLE rfc003_bitcoin_ethereum_bitcoin_erc20_request_messages
+(
+    id INTEGER               NOT NULL PRIMARY KEY,
+    swap_id UNIQUE           NOT NULL,
+    bitcoin_network          NOT NULL,
+    ethereum_chain_id        NOT NULL,
+    bitcoin_amount           NOT NULL,
+    erc20_amount             NOT NULL,
+    erc20_token_contract     NOT NULL,
+    hash_function            NOT NULL,
+    bitcoin_refund_identity  NOT NULL,
+    ethereum_redeem_identity NOT NULL,
+    bitcoin_expiry           NOT NULL,
+    ethereum_expiry          NOT NULL,
+    secret_hash              NOT NULL
+);
+
+CREATE TABLE rfc003_ethereum_bitcoin_erc20_bitcoin_request_messages
+(
+    id INTEGER               NOT NULL PRIMARY KEY,
+    swap_id UNIQUE           NOT NULL,
+    bitcoin_network          NOT NULL,
+    ethereum_chain_id        NOT NULL,
+    bitcoin_amount           NOT NULL,
+    erc20_amount             NOT NULL,
+    erc20_token_contract     NOT NULL,
+    hash_function            NOT NULL,
+    bitcoin_redeem_identity  NOT NULL,
+    ethereum_refund_identity NOT NULL,
+    bitcoin_expiry           NOT NULL,
+    ethereum_expiry          NOT NULL,
+    secret_hash              NOT NULL
+);
+
+CREATE TABLE rfc003_ethereum_bitcoin_accept_messages
+(
+    id INTEGER               NOT NULL PRIMARY KEY,
+    swap_id UNIQUE           NOT NULL,
+    bitcoin_refund_identity  NOT NULL,
+    ethereum_redeem_identity NOT NULL,
+    at DATETIME DEFAULT CURRENT_TIMESTAMP
+);
+
+CREATE TABLE rfc003_bitcoin_ethereum_accept_messages
+(
+    id INTEGER               NOT NULL PRIMARY KEY,
+    swap_id UNIQUE           NOT NULL,
+    bitcoin_redeem_identity  NOT NULL,
+    ethereum_refund_identity NOT NULL,
+    at DATETIME DEFAULT CURRENT_TIMESTAMP
+);
+
+CREATE TABLE rfc003_decline_messages
+(
+    id INTEGER     	NOT NULL PRIMARY KEY,
+    swap_id UNIQUE 	NOT NULL,
+    reason
+);
+
+CREATE TABLE rfc003_swaps
+(
+    id INTEGER     	NOT NULL PRIMARY KEY,
+    swap_id UNIQUE 	NOT NULL,
+    role 		NOT NULL,
+    counterparty 	NOT NULL
+);
+
+CREATE TABLE finalized_swaps
+(
+    id INTEGER          NOT NULL PRIMARY KEY,
+    swap_id UNIQUE      NOT NULL,
```

Comment:
Which one is this swap id? The local or shared one? Either way, I think both should be stored.

Reply:
I'm working on the 'created' side as well.  Please review that one when its ready and lets discuss this over there because I already add a comment dealing with this to `schema.rs`.  Otherwise the conversation is going to be broken/spilt over the two PRs.  Thanks